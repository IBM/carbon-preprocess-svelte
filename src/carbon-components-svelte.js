export const components = {
  "metadata": {
    "package": "carbon-components-svelte",
    "version": "0.62.0"
  },
  "components": {
    "index": {
      "path": "carbon-components-svelte/src/UIShell/index.js"
    },
    "Accordion": {
      "path": "carbon-components-svelte/src/Accordion/Accordion.svelte"
    },
    "AccordionItem": {
      "path": "carbon-components-svelte/src/Accordion/AccordionItem.svelte"
    },
    "AccordionSkeleton": {
      "path": "carbon-components-svelte/src/Accordion/AccordionSkeleton.svelte"
    },
    "Breadcrumb": {
      "path": "carbon-components-svelte/src/Breadcrumb/Breadcrumb.svelte"
    },
    "BreadcrumbSkeleton": {
      "path": "carbon-components-svelte/src/Breadcrumb/BreadcrumbSkeleton.svelte"
    },
    "BreadcrumbItem": {
      "path": "carbon-components-svelte/src/Breadcrumb/BreadcrumbItem.svelte"
    },
    "Breakpoint": {
      "path": "carbon-components-svelte/src/Breakpoint/Breakpoint.svelte"
    },
    "breakpointObserver.d": {
      "path": "carbon-components-svelte/src/Breakpoint/breakpointObserver.d.ts"
    },
    "breakpointObserver": {
      "path": "carbon-components-svelte/src/Breakpoint/breakpointObserver.js"
    },
    "breakpoints.d": {
      "path": "carbon-components-svelte/src/Breakpoint/breakpoints.d.ts"
    },
    "breakpoints": {
      "path": "carbon-components-svelte/src/Breakpoint/breakpoints.js"
    },
    "index.d": {
      "path": "carbon-components-svelte/src/Breakpoint/index.d.ts"
    },
    "AspectRatio": {
      "path": "carbon-components-svelte/src/AspectRatio/AspectRatio.svelte"
    },
    "Button": {
      "path": "carbon-components-svelte/src/Button/Button.svelte"
    },
    "ButtonSet": {
      "path": "carbon-components-svelte/src/Button/ButtonSet.svelte"
    },
    "ButtonSkeleton": {
      "path": "carbon-components-svelte/src/Button/ButtonSkeleton.svelte"
    },
    "CodeSnippet": {
      "path": "carbon-components-svelte/src/CodeSnippet/CodeSnippet.svelte"
    },
    "CodeSnippetSkeleton": {
      "path": "carbon-components-svelte/src/CodeSnippet/CodeSnippetSkeleton.svelte"
    },
    "Checkbox": {
      "path": "carbon-components-svelte/src/Checkbox/Checkbox.svelte"
    },
    "InlineCheckbox": {
      "path": "carbon-components-svelte/src/Checkbox/InlineCheckbox.svelte"
    },
    "CheckboxSkeleton": {
      "path": "carbon-components-svelte/src/Checkbox/CheckboxSkeleton.svelte"
    },
    "ComboBox": {
      "path": "carbon-components-svelte/src/ComboBox/ComboBox.svelte"
    },
    "ComposedModal": {
      "path": "carbon-components-svelte/src/ComposedModal/ComposedModal.svelte"
    },
    "ModalFooter": {
      "path": "carbon-components-svelte/src/ComposedModal/ModalFooter.svelte"
    },
    "ModalHeader": {
      "path": "carbon-components-svelte/src/ComposedModal/ModalHeader.svelte"
    },
    "CopyButton": {
      "path": "carbon-components-svelte/src/CopyButton/CopyButton.svelte"
    },
    "ModalBody": {
      "path": "carbon-components-svelte/src/ComposedModal/ModalBody.svelte"
    },
    "ContentSwitcher": {
      "path": "carbon-components-svelte/src/ContentSwitcher/ContentSwitcher.svelte"
    },
    "Switch": {
      "path": "carbon-components-svelte/src/ContentSwitcher/Switch.svelte"
    },
    "ContextMenuDivider": {
      "path": "carbon-components-svelte/src/ContextMenu/ContextMenuDivider.svelte"
    },
    "ContextMenuGroup": {
      "path": "carbon-components-svelte/src/ContextMenu/ContextMenuGroup.svelte"
    },
    "ContextMenu": {
      "path": "carbon-components-svelte/src/ContextMenu/ContextMenu.svelte"
    },
    "ContextMenuRadioGroup": {
      "path": "carbon-components-svelte/src/ContextMenu/ContextMenuRadioGroup.svelte"
    },
    "ContextMenuOption": {
      "path": "carbon-components-svelte/src/ContextMenu/ContextMenuOption.svelte"
    },
    "DataTable": {
      "path": "carbon-components-svelte/src/DataTable/DataTable.svelte"
    },
    "Table": {
      "path": "carbon-components-svelte/src/DataTable/Table.svelte"
    },
    "TableBody": {
      "path": "carbon-components-svelte/src/DataTable/TableBody.svelte"
    },
    "TableCell": {
      "path": "carbon-components-svelte/src/DataTable/TableCell.svelte"
    },
    "TableContainer": {
      "path": "carbon-components-svelte/src/DataTable/TableContainer.svelte"
    },
    "TableHead": {
      "path": "carbon-components-svelte/src/DataTable/TableHead.svelte"
    },
    "TableHeader": {
      "path": "carbon-components-svelte/src/DataTable/TableHeader.svelte"
    },
    "TableRow": {
      "path": "carbon-components-svelte/src/DataTable/TableRow.svelte"
    },
    "Toolbar": {
      "path": "carbon-components-svelte/src/DataTable/Toolbar.svelte"
    },
    "ToolbarBatchActions": {
      "path": "carbon-components-svelte/src/DataTable/ToolbarBatchActions.svelte"
    },
    "ToolbarContent": {
      "path": "carbon-components-svelte/src/DataTable/ToolbarContent.svelte"
    },
    "ToolbarMenu": {
      "path": "carbon-components-svelte/src/DataTable/ToolbarMenu.svelte"
    },
    "ToolbarMenuItem": {
      "path": "carbon-components-svelte/src/DataTable/ToolbarMenuItem.svelte"
    },
    "ToolbarSearch": {
      "path": "carbon-components-svelte/src/DataTable/ToolbarSearch.svelte"
    },
    "Dropdown": {
      "path": "carbon-components-svelte/src/Dropdown/Dropdown.svelte"
    },
    "DataTableSkeleton": {
      "path": "carbon-components-svelte/src/DataTable/DataTableSkeleton.svelte"
    },
    "DropdownSkeleton": {
      "path": "carbon-components-svelte/src/Dropdown/DropdownSkeleton.svelte"
    },
    "DatePickerInput": {
      "path": "carbon-components-svelte/src/DatePicker/DatePickerInput.svelte"
    },
    "DatePicker": {
      "path": "carbon-components-svelte/src/DatePicker/DatePicker.svelte"
    },
    "createCalendar": {
      "path": "carbon-components-svelte/src/DatePicker/createCalendar.js"
    },
    "DatePickerSkeleton": {
      "path": "carbon-components-svelte/src/DatePicker/DatePickerSkeleton.svelte"
    },
    "FileUploader": {
      "path": "carbon-components-svelte/src/FileUploader/FileUploader.svelte"
    },
    "FileUploaderDropContainer": {
      "path": "carbon-components-svelte/src/FileUploader/FileUploaderDropContainer.svelte"
    },
    "FileUploaderButton": {
      "path": "carbon-components-svelte/src/FileUploader/FileUploaderButton.svelte"
    },
    "FileUploaderItem": {
      "path": "carbon-components-svelte/src/FileUploader/FileUploaderItem.svelte"
    },
    "FileUploaderSkeleton": {
      "path": "carbon-components-svelte/src/FileUploader/FileUploaderSkeleton.svelte"
    },
    "Filename": {
      "path": "carbon-components-svelte/src/FileUploader/Filename.svelte"
    },
    "Form": {
      "path": "carbon-components-svelte/src/Form/Form.svelte"
    },
    "FluidForm": {
      "path": "carbon-components-svelte/src/FluidForm/FluidForm.svelte"
    },
    "FormGroup": {
      "path": "carbon-components-svelte/src/FormGroup/FormGroup.svelte"
    },
    "FormItem": {
      "path": "carbon-components-svelte/src/FormItem/FormItem.svelte"
    },
    "Column": {
      "path": "carbon-components-svelte/src/Grid/Column.svelte"
    },
    "Grid": {
      "path": "carbon-components-svelte/src/Grid/Grid.svelte"
    },
    "Row": {
      "path": "carbon-components-svelte/src/Grid/Row.svelte"
    },
    "InlineLoading": {
      "path": "carbon-components-svelte/src/InlineLoading/InlineLoading.svelte"
    },
    "FormLabel": {
      "path": "carbon-components-svelte/src/FormLabel/FormLabel.svelte"
    },
    "Icon": {
      "path": "carbon-components-svelte/src/Icon/Icon.svelte"
    },
    "IconSkeleton": {
      "path": "carbon-components-svelte/src/Icon/IconSkeleton.svelte"
    },
    "Link": {
      "path": "carbon-components-svelte/src/Link/Link.svelte"
    },
    "OutboundLink": {
      "path": "carbon-components-svelte/src/Link/OutboundLink.svelte"
    },
    "ImageLoader": {
      "path": "carbon-components-svelte/src/ImageLoader/ImageLoader.svelte"
    },
    "LocalStorage": {
      "path": "carbon-components-svelte/src/LocalStorage/LocalStorage.svelte"
    },
    "ListBoxField": {
      "path": "carbon-components-svelte/src/ListBox/ListBoxField.svelte"
    },
    "ListBoxMenu": {
      "path": "carbon-components-svelte/src/ListBox/ListBoxMenu.svelte"
    },
    "ListBoxMenuIcon": {
      "path": "carbon-components-svelte/src/ListBox/ListBoxMenuIcon.svelte"
    },
    "ListBoxSelection": {
      "path": "carbon-components-svelte/src/ListBox/ListBoxSelection.svelte"
    },
    "ListBoxMenuItem": {
      "path": "carbon-components-svelte/src/ListBox/ListBoxMenuItem.svelte"
    },
    "ListItem": {
      "path": "carbon-components-svelte/src/ListItem/ListItem.svelte"
    },
    "Loading": {
      "path": "carbon-components-svelte/src/Loading/Loading.svelte"
    },
    "Modal": {
      "path": "carbon-components-svelte/src/Modal/Modal.svelte"
    },
    "ListBox": {
      "path": "carbon-components-svelte/src/ListBox/ListBox.svelte"
    },
    "NumberInput": {
      "path": "carbon-components-svelte/src/NumberInput/NumberInput.svelte"
    },
    "NumberInputSkeleton": {
      "path": "carbon-components-svelte/src/NumberInput/NumberInputSkeleton.svelte"
    },
    "MultiSelect": {
      "path": "carbon-components-svelte/src/MultiSelect/MultiSelect.svelte"
    },
    "InlineNotification": {
      "path": "carbon-components-svelte/src/Notification/InlineNotification.svelte"
    },
    "NotificationActionButton": {
      "path": "carbon-components-svelte/src/Notification/NotificationActionButton.svelte"
    },
    "NotificationButton": {
      "path": "carbon-components-svelte/src/Notification/NotificationButton.svelte"
    },
    "NotificationIcon": {
      "path": "carbon-components-svelte/src/Notification/NotificationIcon.svelte"
    },
    "NotificationTextDetails": {
      "path": "carbon-components-svelte/src/Notification/NotificationTextDetails.svelte"
    },
    "ToastNotification": {
      "path": "carbon-components-svelte/src/Notification/ToastNotification.svelte"
    },
    "OrderedList": {
      "path": "carbon-components-svelte/src/OrderedList/OrderedList.svelte"
    },
    "PaginationSkeleton": {
      "path": "carbon-components-svelte/src/Pagination/PaginationSkeleton.svelte"
    },
    "Pagination": {
      "path": "carbon-components-svelte/src/Pagination/Pagination.svelte"
    },
    "OverflowMenuItem": {
      "path": "carbon-components-svelte/src/OverflowMenu/OverflowMenuItem.svelte"
    },
    "OverflowMenu": {
      "path": "carbon-components-svelte/src/OverflowMenu/OverflowMenu.svelte"
    },
    "PaginationItem": {
      "path": "carbon-components-svelte/src/PaginationNav/PaginationItem.svelte"
    },
    "PaginationNav": {
      "path": "carbon-components-svelte/src/PaginationNav/PaginationNav.svelte"
    },
    "PaginationOverflow": {
      "path": "carbon-components-svelte/src/PaginationNav/PaginationOverflow.svelte"
    },
    "ProgressBar": {
      "path": "carbon-components-svelte/src/ProgressBar/ProgressBar.svelte"
    },
    "Popover": {
      "path": "carbon-components-svelte/src/Popover/Popover.svelte"
    },
    "RadioButton": {
      "path": "carbon-components-svelte/src/RadioButton/RadioButton.svelte"
    },
    "RadioButtonSkeleton": {
      "path": "carbon-components-svelte/src/RadioButton/RadioButtonSkeleton.svelte"
    },
    "Search": {
      "path": "carbon-components-svelte/src/Search/Search.svelte"
    },
    "SearchSkeleton": {
      "path": "carbon-components-svelte/src/Search/SearchSkeleton.svelte"
    },
    "ProgressIndicatorSkeleton": {
      "path": "carbon-components-svelte/src/ProgressIndicator/ProgressIndicatorSkeleton.svelte"
    },
    "ProgressIndicator": {
      "path": "carbon-components-svelte/src/ProgressIndicator/ProgressIndicator.svelte"
    },
    "RadioButtonGroup": {
      "path": "carbon-components-svelte/src/RadioButtonGroup/RadioButtonGroup.svelte"
    },
    "ProgressStep": {
      "path": "carbon-components-svelte/src/ProgressIndicator/ProgressStep.svelte"
    },
    "RecursiveList": {
      "path": "carbon-components-svelte/src/RecursiveList/RecursiveList.svelte"
    },
    "RecursiveListItem": {
      "path": "carbon-components-svelte/src/RecursiveList/RecursiveListItem.svelte"
    },
    "Select": {
      "path": "carbon-components-svelte/src/Select/Select.svelte"
    },
    "SelectItem": {
      "path": "carbon-components-svelte/src/Select/SelectItem.svelte"
    },
    "SelectItemGroup": {
      "path": "carbon-components-svelte/src/Select/SelectItemGroup.svelte"
    },
    "SelectSkeleton": {
      "path": "carbon-components-svelte/src/Select/SelectSkeleton.svelte"
    },
    "SkeletonPlaceholder": {
      "path": "carbon-components-svelte/src/SkeletonPlaceholder/SkeletonPlaceholder.svelte"
    },
    "SkeletonText": {
      "path": "carbon-components-svelte/src/SkeletonText/SkeletonText.svelte"
    },
    "Slider": {
      "path": "carbon-components-svelte/src/Slider/Slider.svelte"
    },
    "SliderSkeleton": {
      "path": "carbon-components-svelte/src/Slider/SliderSkeleton.svelte"
    },
    "Tag": {
      "path": "carbon-components-svelte/src/Tag/Tag.svelte"
    },
    "TagSkeleton": {
      "path": "carbon-components-svelte/src/Tag/TagSkeleton.svelte"
    },
    "Tab": {
      "path": "carbon-components-svelte/src/Tabs/Tab.svelte"
    },
    "TabContent": {
      "path": "carbon-components-svelte/src/Tabs/TabContent.svelte"
    },
    "TabsSkeleton": {
      "path": "carbon-components-svelte/src/Tabs/TabsSkeleton.svelte"
    },
    "Tabs": {
      "path": "carbon-components-svelte/src/Tabs/Tabs.svelte"
    },
    "TextArea": {
      "path": "carbon-components-svelte/src/TextArea/TextArea.svelte"
    },
    "TextAreaSkeleton": {
      "path": "carbon-components-svelte/src/TextArea/TextAreaSkeleton.svelte"
    },
    "StructuredList": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredList.svelte"
    },
    "StructuredListBody": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListBody.svelte"
    },
    "StructuredListCell": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListCell.svelte"
    },
    "StructuredListInput": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListInput.svelte"
    },
    "StructuredListRow": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListRow.svelte"
    },
    "StructuredListHead": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListHead.svelte"
    },
    "StructuredListSkeleton": {
      "path": "carbon-components-svelte/src/StructuredList/StructuredListSkeleton.svelte"
    },
    "TextInput": {
      "path": "carbon-components-svelte/src/TextInput/TextInput.svelte"
    },
    "TimePicker": {
      "path": "carbon-components-svelte/src/TimePicker/TimePicker.svelte"
    },
    "TextInputSkeleton": {
      "path": "carbon-components-svelte/src/TextInput/TextInputSkeleton.svelte"
    },
    "PasswordInput": {
      "path": "carbon-components-svelte/src/TextInput/PasswordInput.svelte"
    },
    "TimePickerSelect": {
      "path": "carbon-components-svelte/src/TimePicker/TimePickerSelect.svelte"
    },
    "Toggle": {
      "path": "carbon-components-svelte/src/Toggle/Toggle.svelte"
    },
    "ToggleSkeleton": {
      "path": "carbon-components-svelte/src/Toggle/ToggleSkeleton.svelte"
    },
    "ClickableTile": {
      "path": "carbon-components-svelte/src/Tile/ClickableTile.svelte"
    },
    "ExpandableTile": {
      "path": "carbon-components-svelte/src/Tile/ExpandableTile.svelte"
    },
    "RadioTile": {
      "path": "carbon-components-svelte/src/Tile/RadioTile.svelte"
    },
    "SelectableTile": {
      "path": "carbon-components-svelte/src/Tile/SelectableTile.svelte"
    },
    "Tile": {
      "path": "carbon-components-svelte/src/Tile/Tile.svelte"
    },
    "TileGroup": {
      "path": "carbon-components-svelte/src/Tile/TileGroup.svelte"
    },
    "Theme": {
      "path": "carbon-components-svelte/src/Theme/Theme.svelte"
    },
    "TooltipDefinition": {
      "path": "carbon-components-svelte/src/TooltipDefinition/TooltipDefinition.svelte"
    },
    "ToggleSmall": {
      "path": "carbon-components-svelte/src/ToggleSmall/ToggleSmall.svelte"
    },
    "ToggleSmallSkeleton": {
      "path": "carbon-components-svelte/src/ToggleSmall/ToggleSmallSkeleton.svelte"
    },
    "Tooltip": {
      "path": "carbon-components-svelte/src/Tooltip/Tooltip.svelte"
    },
    "TooltipFooter": {
      "path": "carbon-components-svelte/src/Tooltip/TooltipFooter.svelte"
    },
    "TooltipIcon": {
      "path": "carbon-components-svelte/src/TooltipIcon/TooltipIcon.svelte"
    },
    "Truncate": {
      "path": "carbon-components-svelte/src/Truncate/Truncate.svelte"
    },
    "TreeView": {
      "path": "carbon-components-svelte/src/TreeView/TreeView.svelte"
    },
    "TreeViewNodeList": {
      "path": "carbon-components-svelte/src/TreeView/TreeViewNodeList.svelte"
    },
    "TreeViewNode": {
      "path": "carbon-components-svelte/src/TreeView/TreeViewNode.svelte"
    },
    "UnorderedList": {
      "path": "carbon-components-svelte/src/UnorderedList/UnorderedList.svelte"
    },
    "Content": {
      "path": "carbon-components-svelte/src/UIShell/Content.svelte"
    },
    "HeaderGlobalAction": {
      "path": "carbon-components-svelte/src/UIShell/HeaderGlobalAction.svelte"
    },
    "HeaderSearch": {
      "path": "carbon-components-svelte/src/UIShell/HeaderSearch.svelte"
    },
    "SkipToContent": {
      "path": "carbon-components-svelte/src/UIShell/SkipToContent.svelte"
    },
    "SideNavDivider": {
      "path": "carbon-components-svelte/src/UIShell/SideNavDivider.svelte"
    },
    "navStore": {
      "path": "carbon-components-svelte/src/UIShell/navStore.js"
    },
    "searchStore": {
      "path": "carbon-components-svelte/src/UIShell/searchStore.js"
    },
    "Add16": {
      "path": "carbon-components-svelte/src/icons/Add16.svelte"
    },
    "AppSwitcher20": {
      "path": "carbon-components-svelte/src/icons/AppSwitcher20.svelte"
    },
    "Calendar16": {
      "path": "carbon-components-svelte/src/icons/Calendar16.svelte"
    },
    "ArrowUp20": {
      "path": "carbon-components-svelte/src/icons/ArrowUp20.svelte"
    },
    "CaretDown16": {
      "path": "carbon-components-svelte/src/icons/CaretDown16.svelte"
    },
    "ArrowsVertical20": {
      "path": "carbon-components-svelte/src/icons/ArrowsVertical20.svelte"
    },
    "CaretRight16": {
      "path": "carbon-components-svelte/src/icons/CaretRight16.svelte"
    },
    "CaretLeft16": {
      "path": "carbon-components-svelte/src/icons/CaretLeft16.svelte"
    },
    "Checkmark16": {
      "path": "carbon-components-svelte/src/icons/Checkmark16.svelte"
    },
    "CheckmarkFilled16": {
      "path": "carbon-components-svelte/src/icons/CheckmarkFilled16.svelte"
    },
    "CheckmarkFilled20": {
      "path": "carbon-components-svelte/src/icons/CheckmarkFilled20.svelte"
    },
    "CheckmarkOutline16": {
      "path": "carbon-components-svelte/src/icons/CheckmarkOutline16.svelte"
    },
    "ChevronDown16": {
      "path": "carbon-components-svelte/src/icons/ChevronDown16.svelte"
    },
    "ChevronDownGlyph": {
      "path": "carbon-components-svelte/src/icons/ChevronDownGlyph.svelte"
    },
    "ChevronRight16": {
      "path": "carbon-components-svelte/src/icons/ChevronRight16.svelte"
    },
    "CircleDash16": {
      "path": "carbon-components-svelte/src/icons/CircleDash16.svelte"
    },
    "Close20": {
      "path": "carbon-components-svelte/src/icons/Close20.svelte"
    },
    "Copy16": {
      "path": "carbon-components-svelte/src/icons/Copy16.svelte"
    },
    "Close16": {
      "path": "carbon-components-svelte/src/icons/Close16.svelte"
    },
    "ErrorFilled20": {
      "path": "carbon-components-svelte/src/icons/ErrorFilled20.svelte"
    },
    "ErrorFilled16": {
      "path": "carbon-components-svelte/src/icons/ErrorFilled16.svelte"
    },
    "Incomplete16": {
      "path": "carbon-components-svelte/src/icons/Incomplete16.svelte"
    },
    "EditOff16": {
      "path": "carbon-components-svelte/src/icons/EditOff16.svelte"
    },
    "Information16": {
      "path": "carbon-components-svelte/src/icons/Information16.svelte"
    },
    "InformationFilled20": {
      "path": "carbon-components-svelte/src/icons/InformationFilled20.svelte"
    },
    "InformationSquareFilled20": {
      "path": "carbon-components-svelte/src/icons/InformationSquareFilled20.svelte"
    },
    "Launch16": {
      "path": "carbon-components-svelte/src/icons/Launch16.svelte"
    },
    "Menu20": {
      "path": "carbon-components-svelte/src/icons/Menu20.svelte"
    },
    "OverflowMenuHorizontal16": {
      "path": "carbon-components-svelte/src/icons/OverflowMenuHorizontal16.svelte"
    },
    "Search20": {
      "path": "carbon-components-svelte/src/icons/Search20.svelte"
    },
    "Settings16": {
      "path": "carbon-components-svelte/src/icons/Settings16.svelte"
    },
    "OverflowMenuVertical16": {
      "path": "carbon-components-svelte/src/icons/OverflowMenuVertical16.svelte"
    },
    "Search16": {
      "path": "carbon-components-svelte/src/icons/Search16.svelte"
    },
    "Subtract16": {
      "path": "carbon-components-svelte/src/icons/Subtract16.svelte"
    },
    "View16": {
      "path": "carbon-components-svelte/src/icons/View16.svelte"
    },
    "ViewOff16": {
      "path": "carbon-components-svelte/src/icons/ViewOff16.svelte"
    },
    "Warning16": {
      "path": "carbon-components-svelte/src/icons/Warning16.svelte"
    },
    "WarningAltFilled16": {
      "path": "carbon-components-svelte/src/icons/WarningAltFilled16.svelte"
    },
    "WarningAltFilled20": {
      "path": "carbon-components-svelte/src/icons/WarningAltFilled20.svelte"
    },
    "WarningFilled16": {
      "path": "carbon-components-svelte/src/icons/WarningFilled16.svelte"
    },
    "WarningFilled20": {
      "path": "carbon-components-svelte/src/icons/WarningFilled20.svelte"
    },
    "HamburgerMenu": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/HamburgerMenu.svelte"
    },
    "SideNav": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/SideNav.svelte"
    },
    "SideNavMenu": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/SideNavMenu.svelte"
    },
    "SideNavItems": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/SideNavItems.svelte"
    },
    "SideNavMenuItem": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/SideNavMenuItem.svelte"
    },
    "SideNavLink": {
      "path": "carbon-components-svelte/src/UIShell/SideNav/SideNavLink.svelte"
    },
    "Header": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/Header.svelte"
    },
    "HeaderAction": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderAction.svelte"
    },
    "HeaderActionLink": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderActionLink.svelte"
    },
    "HeaderActionSearch": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderActionSearch.svelte"
    },
    "HeaderNavItem": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderNavItem.svelte"
    },
    "HeaderNavMenu": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderNavMenu.svelte"
    },
    "HeaderNav": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderNav.svelte"
    },
    "HeaderPanelDivider": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderPanelDivider.svelte"
    },
    "HeaderPanelLinks": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderPanelLinks.svelte"
    },
    "HeaderPanelLink": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderPanelLink.svelte"
    },
    "HeaderUtilities": {
      "path": "carbon-components-svelte/src/UIShell/GlobalHeader/HeaderUtilities.svelte"
    }
  }
}