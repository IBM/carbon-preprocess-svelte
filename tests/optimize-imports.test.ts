import { optimizeImports } from "@";
import type { Preprocessor, Processed } from "svelte/types/compiler/preprocess";

const preprocess = (options?: Partial<Parameters<Preprocessor>[0]>) => {
  return (
    optimizeImports().script({
      attributes: {},
      filename: "test.svelte",
      content: "",
      markup: "",
      ...options,
    }) as Processed
  )?.code;
};

describe("optimizeImports", () => {
  test("preprocessor is skipped", () => {
    expect(preprocess({ filename: undefined })).toBeUndefined();
    expect(preprocess({ filename: "node_modules" })).toBeUndefined();
    expect(
      preprocess({ filename: "node_modules/carbon-components-svelte" }),
    ).toBeUndefined();
  });

  test("barrel imports", () => {
    expect(
      preprocess({
        content: `import { Accordion, AccordionItem } from "carbon-components-svelte";
import { Accordion as Accordion2 } from "carbon-components-svelte";
import { breakpoints } from "carbon-components-svelte";
import { invalid } from "carbon-components-svelte";

import { Add } from "carbon-icons-svelte";
import { Add as Add2 } from "carbon-icons-svelte";
import Add3 from "carbon-icons-svelte/lib/Add.svelte";

import { Airplane } from "carbon-pictograms-svelte";
import { Airplane as Airplane2 } from "carbon-pictograms-svelte";
import Airplane3 from "carbon-pictograms-svelte/lib/Airplane.svelte";`,
      }),
    )
      .toEqual(`import Accordion from "carbon-components-svelte/src/Accordion/Accordion.svelte";import AccordionItem from "carbon-components-svelte/src/Accordion/AccordionItem.svelte";
import Accordion2 from "carbon-components-svelte/src/Accordion/Accordion.svelte";
import breakpoints from "carbon-components-svelte/src/Breakpoint/breakpoints.js";
import { invalid } from "carbon-components-svelte";

import Add from "carbon-icons-svelte/lib/Add.svelte";
import Add2 from "carbon-icons-svelte/lib/Add.svelte";
import Add3 from "carbon-icons-svelte/lib/Add.svelte";

import Airplane from "carbon-pictograms-svelte/lib/Airplane.svelte";
import Airplane2 from "carbon-pictograms-svelte/lib/Airplane.svelte";
import Airplane3 from "carbon-pictograms-svelte/lib/Airplane.svelte";`);
  });
});
