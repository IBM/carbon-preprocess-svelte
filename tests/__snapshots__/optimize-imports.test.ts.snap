// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`optimizeImports barrel imports 1`] = `
"
        import Accordion from "carbon-components-svelte/src/Accordion/Accordion.svelte";import AccordionItem from "carbon-components-svelte/src/Accordion/AccordionItem.svelte";
        import Accordion2 from "carbon-components-svelte/src/Accordion/Accordion.svelte";
        import breakpoints from "carbon-components-svelte/src/Breakpoint/breakpoints.js";
        import { invalid } from "carbon-components-svelte";

        import Add from "carbon-icons-svelte/lib/Add.svelte";
        import Add2 from "carbon-icons-svelte/lib/Add.svelte";
        import Add3 from "carbon-icons-svelte/lib/Add.svelte";

        import Airplane from "carbon-pictograms-svelte/lib/Airplane.svelte";
        import Airplane2 from "carbon-pictograms-svelte/lib/Airplane.svelte";
        import Airplane3 from "carbon-pictograms-svelte/lib/Airplane.svelte";
    "
`;
